src/main/java/com/github/forax/umldoc/gen/MermaidGenerator.java

```diff
@@ -119,6 +121,67 @@ private static void generateEnumConstants(List<Field> fields, Writer writer)
     }
   }
 
+  private static String groupKind(Call.Group.Kind groupKind) {
+    return switch (groupKind) {
+      case LOOP -> "loop\n";
+      case OPTIONAL -> "opt\n";
+      case ALTERNATE -> "alt\n";
+      case PARALLEL -> "par\n";
+      default -> "";
+    };
+  }
+
+  private static void generateMethodCall(String sourceEntity, String dstEntity, String methodName,
+                                         Writer writer) throws IOException {
+
+    writer.append(sourceEntity)
+            .append(" ->> ")
+            .append(dstEntity)
+            .append(": ")
+            .append(methodName)
+            .append("()\n");
+  }
+
+  private static void generateCalls(Call.Group group, Set<Entity> entities,
+                                    Writer writer) throws IOException {
+    if (group.equals(Call.Group.EMPTY_GROUP)) {
+      return;
+    }
+
+    var currentGroupKind = group.kind();
+
+    if (!currentGroupKind.equals(Call.Group.Kind.NONE)) {
+      writer.append(groupKind(currentGroupKind));
+    }
+
+    var nbCalls = 0;
+    for (var call : group.calls()) {
+      nbCalls += 1;
+      if (call instanceof Call.Group groupCall) {
+        generateCalls(
+                groupCall,
+                entities,
+                writer
+        );
+
+        if (currentGroupKind.equals(Call.Group.Kind.ALTERNATE) && nbCalls < group.calls().size()) {
+          writer.append("else\n");
+        }
+      }
+
+      if (call instanceof Call.MethodCall methodCall) {
```

Comment:
you should use if ... else and throw an exception if it's not one of the possible subtypes